rotkehlchen/api/rest.py

```diff
@@ -534,13 +535,22 @@ def _get_trades(
 
         trades_result = []
         for trade in trades:
-            serialized_trade = self.trade_schema.dump(trade)
-            serialized_trade['trade_id'] = trade.identifier
+            if isinstance(trade, AMMTrade):
+                serialized_trade = trade.serialize()
+            else:
+                serialized_trade = self.trade_schema.dump(trade)
+                serialized_trade['trade_id'] = trade.identifier
             trades_result.append(serialized_trade)
 
+        entry_table: Literal['amm_trades', 'trades']
```

Comment:
Why?

Reply:
Regarding the logic:
So by ALL you mean `trades` + `amm_trades`? If so it seems I've missed to query all trades for AMMTrade-related locations here (case `location is None`):
https://github.com/rotki/rotki/blob/1f269dca759189225abbab7d391b5fda03f41163/rotkehlchen/rotkehlchen.py#L589

Regarding the typing:
First time with `Literal` type. Mypy warned me if I did `entry_table = 'trades' if ... else 'amm_trades'`, and pass `entry_table` into the count function (which has the cases as `Literal`). It the right way to do it would be with `Final`, as per:
https://mypy.readthedocs.io/en/stable/literal_types.html#declaring-literal-variables
https://stackoverflow.com/questions/63891789/can-mypy-track-string-literals